agent/agent.go

```diff
@@ -487,10 +488,21 @@ func (a *Agent) Publisher(ctx context.Context, subscriptionID string) (exec.LogP
 		return nil, nil, err
 	}
 
+	// make little closure for ending the log stream
+	sendCloseMsg := func() {
+		// send a close message, to tell the manager our logs are done
+		publisher.Send(&api.PublishLogsMessage{
```

Comment:
I'm not entirely sure SendAndClose can be used to close out a stream? It seemed to me like it was only for sending back the response to a stream. But I'll try.